### ISSUE TITLE
Search functionality

### ISSUE THREAD

**Issue by amanda-phet** [phetsims/website#228]
We are wondering if it is a quick project to update one aspect of our current functionality... Right now, if you start typing a sim name into the search bar, it will auto-complete with the sim(s) that match the search:
![screen shot 2015-06-10 at 3 38 06 pm](https://cloud.githubusercontent.com/assets/7852916/8094880/20760082-0f87-11e5-93f4-387650bd7678.png)

However, the search much match the sim name *exactly*. It seems like it should search any word in the sim name. In the example above, "Build a Fraction" is not included in the auto-complete, but it should be. 

Is this possibly a simple fix?

**Comment by aaronsamuel137** [comment]
It's possible to do this, but it doesn't look like it would be simple. I'd guess at least a couple of hours, possibly more depending on what I find as I start looking into it.

**Comment by amanda-phet** [comment]
It is something Michael could work on?

**Comment by aaronsamuel137** [comment]
Yep. Feel free to look into this @zepumph.

You'll want to look at Autocomplete.java. You can test what is being returned by looking at: 

view-source:http://phet.colorado.edu/autocomplete?q=fraction&l=en

which currently outputs:
```
Fraction Matcher|sim|/en/simulation/fraction-matcher
Fractions Intro|sim|/en/simulation/fractions-intro
```

As noted by @amanda-phet, we'd want this to have fraction matcher in the results too.

**Comment by amanda-phet** [comment]
>As noted by @amanda-phet, we'd want this to have fraction matcher in the results too.
Correction: we'll want this to have *Build a fraction* in the results!

**Comment by amanda-phet** [comment]
Now that @zepumph is going to work with us again, perhaps he can work on this item? 

I looked into [google search] (https://support.google.com/customsearch/answer/4513751?hl=en&ref_topic=4513742) a bit and it seems really easy to [implement] (https://support.google.com/customsearch/answer/4513903?hl=en&ref_topic=4513742). 

[Try some searches] (https://cse.google.com/cse/publicurl?cx=011512963771032054849:57h9p8hzhz0) now! @oliver-phet @ariel-phet @amyh-phet @arouinfar  

It normally [costs money] (https://support.google.com/customsearch/answer/4542048?hl=en&ref_topic=4513870) to disable ads, unless you are a [non-profit] (https://support.google.com/customsearch/answer/4542102?hl=en&ref_topic=4513870 ).

@ariel-phet can you think about the priority of this item? If we are ok with using google, this would be so simple.

**Comment by mattpen** [comment]
@amanda-phet - I implemented the original request tonight using our existing toolset.  You can see it in action at https://phet-new.colorado.edu.  This is our new server that will be going online soon.  It should match the query "Frac" with "Build a Fraction" and the query "Base" with "Acid-Base Solutions"

Is this sufficient, or would you prefer to have the full Google search implemented instead?  

Personally, I would recommend against using a custom google search for two reasons.  A) It is easy to miss communication from google about changes to the service. We have used Google services (automatic translation) in the past that just stopped working with no warning. The custom search seems pretty stable, but if we can implement this easily on our own it would be more reliable. B) I'm usually not impressed with websites that use the custom google search.  It normally seems to be less effective than using either the native website search or just using Google as a stand alone service.  I may not be in the majority opinion here though.  On the other side of the coin, if we eventually want to add a lot of features to the search and autocomplete it may be better to incorporate Google now rather than slowly reinvent the wheel.

**Comment by ariel-phet** [comment]
@mattpen and @amanda-phet I think we would also want to discuss with @jonathanolson - he might have reasons for/against

**Comment by arouinfar** [comment]
@amanda-phet I noticed that the Google custom search retrieves results that we wouldn't necessarily want displayed (like directories). These results were at the top of the 2nd page of results when I searched "fraction". 

![image](https://cloud.githubusercontent.com/assets/8419308/16777620/763c8df4-4827-11e6-8ece-15e038d76c2f.png)

**Comment by amanda-phet** [comment]
The thing that prompted me to think about this was when I was adding keywords for a new sim. The fact that these are "hard coded" just seems a little bit strange when a true search engine should mine the text on a sim page and the metadata. So, while there might be good reasons not to use Google, it might be difficult to make an equally powerful search engine ourselves. 

Since @mattpen already worked on this can we at least test the change and use it in the meantime?

**Comment by amanda-phet** [comment]
@arouinfar good point.

**Comment by kathy-phet** [comment]
Just to chime in. Keywords are good to have as metadata though, for companies.

**Comment by amanda-phet** [comment]
@kathy-phet of course. 

Our current search functionality just treats "Line" and "Lines" (for example) as two different keywords when they should give the same search results.

**Comment by kathy-phet** [comment]
Ah, yes, I see.

**Comment by ariel-phet** [comment]
@mattpen said 

Here are the top twenty search terms for the last 4 full weeks of May.  I used May because school was still in session.  This is counts of hits to the /search page according to Google Analytics.  If you need a more formal report, let me know.  The top search term is intentionally empty.

Here are the top twenty search terms for the last 4 full weeks of May.  I used May because school was still in session.  This is counts of hits to the /search page according to Google Analytics.  If you need a more formal report, let me know.  The top search term is intentionally empty.

![image](https://cloud.githubusercontent.com/assets/4051174/16789757/71bc0e94-486d-11e6-95ad-85457629ac8a.png)

**Comment by amanda-phet** [comment]
@mattpen can we go ahead and deploy this? Seems to be working well.

**Comment by amanda-phet** [comment]
>Do you want to continue investigating the Google search? 

Yes, we do want to continue to look into it, but we are still in the process of investigating whether we can qualify for non-profit pricing.

**Comment by mattpen** [comment]
@amanda-phet - The new autocomplete function has been deployed to figaro.

**Comment by amanda-phet** [comment]
Great! I'll keep myself assigned as we continue to investigate google search.

**Comment by ariel-phet** [comment]
Also assigning myself to check in with @jonathanolson if he sees any downsides to the google search option
